### ISSUE TITLE
Too much memory used XXX mb > 40 mb, removing oldest flight ./flight

### ISSUE THREAD

**Issue by RiseOoi** [AutonomyLab/ardrone_autonomy#172]
Successfully ran and manually controlled ardrone 2.0 on Ubuntu 14.04 with ROS indigo for a few flights. Latest version of firmware.

I tried again after an hour later and things start to get strange.

```
...
...
Too much memory used 394 mb > 40 mb, removing oldest flight ./flight 
Too much memory used 394 mb > 40 mb, removing oldest flight ./flight
Too much memory used 394 mb > 40 mb, removing oldest flight ./flight
Too much memory used 394 mb > 40 mb, removing oldest flight ./flight
Too much memory used 394 mb > 40 mb, removing oldest flight ./flight
```

Kept looping the same thing, Ctrl-C could not interrupt and end the loop, must use terminate terminal

Tried resetting the drone and restarting.

Ardrone_Autonomy: commit f7c66282c1f58b2cf8f33a30bb962242aa421c9b on Nov 4 (latest)
ROS version -d : indigo 1.11.13 (most updated)
Firmware : 2.4.8 (latest)
Ubuntu : Trusty 14.04 LTS (most updated in LTS)
Not running Ubuntu inside a virtual machine.
Hardware is average i3 Ivybridge with 4GB of RAM.

I wonder what kind of issue is this.
Any help is greatly appreciated.

**Edit:** ~~It is working fine now.~~ Weird.
**Edit:** It comes out here and then often. A catkin_make (without any changes) solves the problem. (In case if anyone is having a similar problem. You will need to change something that affects ROS like a catkin_make)

**Comment by mani-monaj** [comment]
Please provide more information about your setup, like the firmware version and whether if you are running Ubuntu inside a virtual machine.

**Comment by RiseOoi** [comment]
I have added information of ROS -> indigo and ardrone_autonomy to Nov 4 commit f7c66282c1f58b2cf8f33a30bb962242aa421c9b.

It seems like the too much memory problem always come out after I close a terminal of ardrone_autonomy. The next time I run ardrone_autonomy, no matter restarted my computer or not, the problem will occur. A catkin_make seems to "solve" the problem. Even if nothing is changed in src.

**Comment by mani-monaj** [comment]
That is strange. By firmware version, I mean the firmware version of your AR-Drone 2.0. You can either find that using Parrot's app or by looking at the output of the driver.

**Comment by RiseOoi** [comment]
It is 2.4.8 by $telnet and $cat version.txt. I tried to update it through AR.Freeflight on iOS and I believe both the app and the firmware is on the latest version.
I did some testings yesterday and this morning (That's why it took me so long to reply) and yes the problem is a consistent one. But I have to admit that my insufficient knowledge of ROS maybe the main cause. (Bad catkin declaration / ROS calls). I will look more into it by Monday.

P.S. Thank you for your work. I admire it a lot.
